contracts/core/liquidators/twap-impl/TWAPAuction.sol

```diff
@@ -39,4 +56,357 @@ contract TWAPAuction is TwoAssetPriceBoundedLinearAuction {
+        uint256 usdChunkSize;
+        uint256 chunkAuctionPeriod;
+    }
+
+    /* ============ Constants ============ */
+
+    // Auction completion buffer assumes completion potentially 2% after fair value when auction started
+    uint256 constant AUCTION_COMPLETION_BUFFER = 2;
+
+    /* ============ State Variables ============ */
+
+    mapping(bytes32 => BoundsLibrary.Bounds) public chunkSizeWhiteList;
+    uint256 public expectedChunkAuctionLength;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * TWAPAuction constructor
+     *
+     * @param _oracleWhiteList        OracleWhiteList used by liquidator
+     * @param _auctionPeriod          Length of auction
+     * @param _rangeStart             Percentage below FairValue to begin auction at
+     * @param _rangeEnd               Percentage above FairValue to end auction at
+     * @param _assetPairHashes        Identifying hashes of asset pairs
+     * @param _assetPairBounds        Chunk size bounds for each asset pair
+     */
+    constructor(
+        IOracleWhiteList _oracleWhiteList,
+        uint256 _auctionPeriod,
+        uint256 _rangeStart,
+        uint256 _rangeEnd,
+        bytes32[] memory _assetPairHashes,
+        BoundsLibrary.Bounds[] memory _assetPairBounds
+    )
+        public
+        TwoAssetPriceBoundedLinearAuction(
+            _oracleWhiteList,
+            _auctionPeriod,
+            _rangeStart,
+            _rangeEnd
+        )
+    {
+        require(
+            _assetPairHashes.length == _assetPairBounds.length,
+            "TWAPAuction: Must be equal amount of asset pair hashes and bounds."
+        );
+
+        for (uint8 i = 0; i < _assetPairHashes.length; i++) {
+            require(
+                _assetPairBounds[i].isValid(),
+                "TWAPAuction.constructor: Passed asset pair bounds are invalid."
+            );
+
+            chunkSizeWhiteList[_assetPairHashes[i]] = _assetPairBounds[i];
+        }
+
+        expectedChunkAuctionLength = _auctionPeriod
+            .mul(_rangeStart.add(AUCTION_COMPLETION_BUFFER))
+            .div(_rangeStart.add(_rangeEnd));
+    }
+
+    /* ============ Internal Functions ============ */
+
+    /**
+     * Populates the TWAPState struct and initiates first chunk auction.
+     *
+     * @param _twapAuction                  TWAPAuction State object
+     * @param _currentSet                   The Set to rebalance from
+     * @param _nextSet                      The Set to rebalance to
+     * @param _startingCurrentSetQuantity   Quantity of currentSet to rebalance
+     * @param _liquidatorData               Struct of parsed liquidator data
+     */
+    function initializeTWAPAuction(
+        TWAPState storage _twapAuction,
+        ISetToken _currentSet,
+        ISetToken _nextSet,
+        uint256 _startingCurrentSetQuantity,
+        TWAPLiquidatorData memory _liquidatorData
+    )
+        internal
+    {
+        // Calculate value of rebalance volume
```

Comment:
What is the volume denominated in? Would be useful if you can add it to the comment

Reply:
I guess in theory it could be anything that's why I didn't specify. Perhaps makes the `usdChunkSize` param a misnomer